pkg/controller/associated/index.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package associated handles the iOS and Android associated app handler
+// protocols. For more discussion of these protocols, please see:
+//
+// Android:
+//   https://developer.android.com/training/app-links/verify-site-associations
+// iOS:
+//   https://developer.apple.com/documentation/safariservices/supporting_associated_domains
+package associated
+
+import (
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/google/exposure-notifications-verification-server/pkg/api"
+	"github.com/google/exposure-notifications-verification-server/pkg/cache"
+	"github.com/google/exposure-notifications-verification-server/pkg/controller"
+	"github.com/google/exposure-notifications-verification-server/pkg/database"
+	"github.com/google/exposure-notifications-verification-server/pkg/render"
+)
+
+type Controller struct {
+	ctxt   context.Context
+	db     *database.Database
+	cacher cache.Cacher
+	h      *render.Renderer
+}
+
+var ttl = 30 * time.Minute
+
+func (c *Controller) HandleIos() http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		key := r.URL.String()
+
+		// Check the cache.
```

Comment:
You should be able to use `c.cacher.Fetch` here instead of a manual read, create, write. 

```go
var data IOSData
if err := c.cacher.Fetch(ctx, key, &data, ttl, func() (interface{}, error){
  return c.getIosData()
}); err != nil {
  // handle error
}
```